libevmasm/ExpressionClasses.cpp

```diff
@@ -184,6 +184,7 @@ string ExpressionClasses::fullDAGToString(ExpressionClasses::Id _id) const
 ExpressionClasses::Id ExpressionClasses::tryToSimplify(Expression const& _expr)
 {
 	static Rules rules;
+	assertThrow(rules.isInitialized(), OptimizerException, "Rule list not properly initialized.");
```

Comment:
Is this really needed if in the `Rules` constructor this is done?

Reply:
It shouldn't but better safe than sorry.